src/DistinguishedName.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+/*
+ * This file is part of the ACME PHP library.
+ *
+ * (c) Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace AcmePhp\Ssl;
+
+use Webmozart\Assert\Assert;
+
+/**
+ * Represent a Distinguished Name.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class DistinguishedName
+{
+    /** @var string */
+    private $commonName;
+
+    /** @var string */
+    private $countryName;
+
+    /** @var string */
+    private $stateOrProvinceName;
+
+    /** @var string */
+    private $localityName;
+
+    /** @var string */
+    private $organizationName;
+
+    /** @var string */
+    private $organizationalUnitName;
+
+    /** @var string */
+    private $emailAddress;
+
+    /** @var array */
+    private $subjectAlternativeNames;
+
+    /**
+     * @param string $commonName
+     * @param string $countryName
+     * @param string $stateOrProvinceName
+     * @param string $localityName
+     * @param string $organizationName
+     * @param string $organizationalUnitName
+     * @param string $emailAddress
+     * @param array  $subjectAlternativeNames
+     */
+    public function __construct(
+        $commonName,
```

Comment:
According to https://en.wikipedia.org/wiki/Certificate_signing_request#Structure, `$subjectAlternativeNames` is not required (I think we should set it as empty array by default) but all other fields are.